src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -216,9 +216,25 @@ public void Initialize()
             var path = Configuration.Core.DataDirectory.Combine("System");
             var storeAlertForLateRaise = new List<AlertRaised>();
 
-            var options = Configuration.Core.RunInMemory
-                ? StorageEnvironmentOptions.CreateMemoryOnly()
-                : StorageEnvironmentOptions.ForPath(path.FullPath);
+            StorageEnvironmentOptions options;
+            if (Configuration.Core.RunInMemory)
+            {
+                options = StorageEnvironmentOptions.CreateMemoryOnly();
+            }
+            else
+            {
+                options = StorageEnvironmentOptions.ForPath(path.FullPath);
+                var secretKey = Path.Combine(path.FullPath, "secret.key.encrypted");
+                if (File.Exists(secretKey))
```

Comment:
The file is there, but isn't readable.
The file is readable, but we can't decrypt it.

Etc, etc.

We need to make sure that the user gets a GOOD error message here.

Consider the case of moving a database between machines, what kind of error is the user going to get?

"Failure" <-- I know the reason, but won't tell you?

The user gets this feeling: 

😛😛😛😛😛😛😛😛😛😛😛😛😛😛😛😛😛